Client/Frontend/Autofill/CreditCard/ViewComponents/CreditCardInput/CreditCardInputViewModel.swift

```diff
@@ -0,0 +1,95 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import Foundation
+
+class CreditCardInputViewModel: ObservableObject {
+    let fieldType: CreditCardInputType
+    let fieldHeadline: String
+    let errorString: String
+    var formattedTextLimit: Int
+    var delimiterCharacter: String?
+    var keyboardType: UIKeyboardType
+
+    @Published var isValid = true
+
+    init(fieldType: CreditCardInputType) {
+        self.fieldType = fieldType
+
+        switch fieldType {
+        case .name:
+            self.fieldHeadline = .CreditCard.EditCard.NameOnCardTitle
+            self.errorString = .CreditCard.ErrorState.NameOnCardSublabel
+            self.delimiterCharacter = nil
+            self.formattedTextLimit = 100 // TBA
+            self.keyboardType = .alphabet
+        case .number:
+            self.fieldHeadline = .CreditCard.EditCard.CardNumberTitle
+            self.errorString = .CreditCard.ErrorState.CardNumberSublabel
+            self.formattedTextLimit = 20
+            self.delimiterCharacter = " "
+            self.keyboardType = .numberPad
+        case .expiration:
+            self.fieldHeadline = .CreditCard.EditCard.CardExpirationDateTitle
+            self.errorString = .CreditCard.ErrorState.CardExpirationDateSublabel
+            self.formattedTextLimit = 7
+            self.delimiterCharacter = " / "
+            self.keyboardType = .numberPad
+        case .securityCode: // Design feedback needed
+            self.fieldHeadline = "Security Code"
+            self.errorString = "Enter the three digit security code"
+            self.formattedTextLimit = 3
+            self.delimiterCharacter = nil
+            self.keyboardType = .numberPad
+        }
+    }
+
+    func updateNameValidity(userInputtedText: String) {
+        isValid = !userInputtedText.isEmpty
+    }
+
+    func updateCardNumberValidity(userInputtedText: String) {
+        isValid = userInputtedText.count >= 15 && userInputtedText.count <= 16
+    }
+
+    func updateExpirationValidity(userInputtedText: String) {
+        let year = Calendar(identifier: .gregorian).dateComponents([.year], from: Date()).year ?? 2023
```

Comment:
We should hard code the current year (2023) here. We should always fall back to the current year based on the date of the system.